src/twisted/mail/interfaces.py

```diff
@@ -0,0 +1,1070 @@
+        @param listener: An object to add to the set of those which will be
+            notified when the contents of this mailbox change.
+        """
+
+
+    def removeListener(listener):
+        """
+        Remove a mailbox change listener
+
+        @type listener: Any object previously added to and not removed from
+            this mailbox as a listener.
+        @param listener: The object to remove from the set of listeners.
+
+        @raise ValueError: Raised when the given object is not a listener for
+            this mailbox.
+        """
+
+
+    def addMessage(message, flags=(), date=None):
+        """
+        Add the given message to this mailbox.
+
+        @type message: A file-like object
+        @param message: The RFC822 formatted message
+
+        @type flags: Any iterable of C{str}
+        @param flags: The flags to associate with this message
+
+        @type date: C{str}
+        @param date: If specified, the date to associate with this message.
+
+        @rtype: C{Deferred}
+        @return: A deferred whose callback is invoked with the message id if
+            the message is added successfully and whose errback is invoked
+            otherwise.
+
+        @raise ReadOnlyMailbox: Raised if this Mailbox is not open for
+            read-write.
+        """
+
+
+    def expunge():
+        """
+        Remove all messages flagged \\Deleted.
+
+        @rtype: C{list} or C{Deferred}
+        @return: The list of message sequence numbers which were deleted, or a
+            C{Deferred} whose callback will be invoked with such a list.
+
+        @raise ReadOnlyMailbox: Raised if this Mailbox is not open for
+            read-write.
+        """
+
+
+    def fetch(messages, uid):
+        """
+        Retrieve one or more messages.
+
+        @type messages: C{MessageSet}
+        @param messages: The identifiers of messages to retrieve information
+            about
+
+        @type uid: C{bool}
+        @param uid: If true, the IDs specified in the query are UIDs; otherwise
+            they are message sequence IDs.
+
+        @rtype: Any iterable of two-tuples of message sequence numbers and
+            implementors of C{IMessageIMAP}.
+        """
+
+
+    def store(messages, flags, mode, uid):
+        """
+        Set the flags of one or more messages.
+
+        @type messages: A MessageSet object with the list of messages requested
+        @param messages: The identifiers of the messages to set the flags of.
+
+        @type flags: sequence of C{str}
+        @param flags: The flags to set, unset, or add.
+
+        @type mode: -1, 0, or 1
+        @param mode: If mode is -1, these flags should be removed from the
+            specified messages. If mode is 1, these flags should be added to
+            the specified messages. If mode is 0, all existing flags should be
+            cleared and these flags should be added.
+
+        @type uid: C{bool}
```

Comment:
Take a pass through the docstring for what you have changed and change:

C{bool} -> L{bool}
C{str} -> L{str}
C{dict} -> L{dict}
C{tuple} -> L{tuple}

etc.